### PR TITLE
Bump react-native-macos version

### PR DESCRIPTION
**Pull request by jfedak**
## Summary

This PR bumps `react-native-macos` version from 0.71 to 0.72. It also features some minor changes that unifies `MacOSExample` and `Example` apps.

## Test plan

Tested with MacOSExample app.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,7 +155,7 @@
     "react-native": "0.72.3",
     "react-native-builder-bob": "^0.18.3",
     "react-native-gesture-handler": "^2.12.0",
-    "react-native-macos": "^0.71.0-0",
+    "react-native-macos": "^0.72.0",
```
Comment (tjzel) [inline package.json:158]: Is this dependency actually needed in the root of our repo?

File: MacOSExample/metro.config.js [hunk 2]
```diff
@@ -25,13 +33,6 @@ module.exports = {
       return acc;
     }, {}),
   },
-
-  transformer: {
```
Comment (tjzel) [inline MacOSExample/metro.config.js:29]: Why do we drop this setting? Is it included in the default config?

**Review by tjzel — CHANGES_REQUESTED** [review]
Looks good to merge, but please answer my questions first 😅

File: MacOSExample/metro.config.js (hunk 2 above)
Comment (jfedak) [reply MacOSExample/metro.config.js:29]: Exactly, it is present in [default config](https://github.com/facebook/react-native/blob/5323221d1442d1573bc65daff618478cb6f056f0/packages/metro-config/index.js)

File: package.json (hunk 1 above)
Comment (jfedak) [reply package.json:158]: It turned out it is not. I'm about to push appropriate changes.

**Review by tjzel — APPROVED** [review]
Me likes

**Review by piaskowyk — APPROVED** [review]
👏

**Pull request merged**